### ISSUE TITLE
v3.7.7+ Breaks Astral Power Tracker

### ISSUE THREAD

**Issue by justinryder** [WeakAuras/WeakAuras2#3460]
### Description

In version 3.7.7 and later, my astral power tracker stopped working. It's been working all of Shadowlands (wrote it in beta) up thought 3.7.6. I checked the [3.7.7 diff](https://github.com/WeakAuras/WeakAuras2/compare/3.7.6...3.7.7), but didn't notice anything that seemed like it would stop this from updating.

This is the full weak aura: https://wago.io/XrU9NuZCX
The specific aura in it affected by 3.7.7+ is called "Astral Power" (I've also included a string export of just the AP bar below).

### WeakAuras Version

WeakAuras 3.7.8

### World of Warcraft Flavor

Retail (Default)

### Tested with only WeakAuras

- [ ] Yes
- [X] No

### Reproduction Steps

1) Load my Weak Aura in v3.7.6 or earlier on a Balance Druid and build some Astral Power
2) The AP bar will show current/predicted AP
3) Load my Weak Aura in v3.7.7 or later on a Balance Druid and build some Astral Power
4) The AP bar will only update when you close the /wa panel but not on cast or AP change

### Last Good Version

WeakAuras 3.7.6

### Export String

!WA:2!1Mt3VTT11hw9qrvrbC06ms6k2ymqtT7CuKK)ijfW5xLSLILRJKmLC(OnzsuKxjYgksUljTLtrgq9kgYE1p87H9202td7j9CrArWETaxyuG9U)lyiV1N2o37LuIuFe10UrGOqEp3Z3N75JRfkKOtc1eQp9A9CSLXp6U6QUA)8NZEVCRwoi37j08q52wf36K7H3)gL8(KnVhUlhKqpfphxRo1qDDTVzlptfxDltXfxseJC9WMIUwoUyDZ2lk7HLRJmpiPJRSlY5tt)WKOU2ifxKAw7LerMQ4J40C7N9Voh7PVSPIMfUILUPBZnZxQwEPtzILKfqdGpc2NpY332sfLl7(1kNXdBy)UAUU2oF4vVkv4tQBD1bs)vtVYlAkR8O2ylpt1nTmSW7iiiujwQKRDmiUTBJWoLUm2)1NEHJvbPNYHePpTnYeH1vQQzDyzZZCy)3MwwgQwhAkzk3b5e)mhVMOdqMUvWOw6DLQwj)U72KBP6k8pdmtlYnfllY27YIEM6WV1xw0XgzyuuDP4IWJEloCXn2qCHQ1YkvBbrxnKjdi9zGL1YSQRm21NSCSrgoOiuy)sfRvNjqBMTk8gNGYMQm2Z2ITH8ri805YMYoUH50ic8Szz5k)O4OL9RcdRu(U5LQxqk)E7dHoVImSI1Hi8(2Qa3IAonvz)p7NGqCShkoaONkQPxRw1oYgH3o)UvkS)U2N3ZbvpsetJtyYEjiqrOppKOUlGsFgB8ySmhDbYfp2xldfqvfyGE3(1h678pasP3RDkgjBuDa5LyOyFXGOtXkyR2yKJJ4ISnTua)10vb2Dg)RT8d1xiE8tPs)aXTbxeCSVlJXItks6d5gwXj4YJakQZXhe2ZKXHCYEUwskAiLhrE)CuFwt(goHkpUy4KBJxWf27iJ1LBAGC6k89FoZPiRQQtfFzJbA7gIzwEOx7azdpeS2NpWVtT2Wcly61PjeuS8aaQ6ougtHTPhgtdVYcPZKnezXh(78jHiUlKjY4vM43wU7SiSUPIvhipAw7xzQx0h1zXIHPJFLzrEFuNflSoaH)rq(YaAkY2tH6p5ept)m1ZtUie22hq9Zy5ypaLr28i7ZlZEVgFt0Qe)9Z1txXYSUcl3Fm45fqrgegcBUdK7hIFqN4GmAXQ9KbIEpMk6G448b7S9)33920NudEjgRYXoXeeIDCtzmVGYe2wmifHJSl9agslCf27l0doHlHAdm25pnp5sNcFoScL)cTSWi(cpDUx4QR8O6qKVPA9oGcLj7wBrUeg2ffWZzqdQrZ3BiwBph9ewh4yUsDGcTD10oJThxn4xt4mZBy)ZylaEafuhiWNXKtY6wb8eWN9zNeHI(EyKwp2w9nLcCJXzrrFHvsXLPd0D0bPUb)l)k(c8V8PN9IfPUIwaQpGM4hcDZjJFW(fVYWVUsvQXRpdnUgSh)Jo6ySfw7u2h4G(eE6CK8KcKlrUfzBbsrnYoAKp(ni7sUnPKgP8ocX4snPYcRLISxdIKaPkPgz)9i3rJC3zIFGwd4V(pr8VXeX)c2xIADyMO6qCa47QBx3vVdQUngPOtJyJDYGTSW7zBFXjGa)fjkEQNXaniiN8wGlNUYGqdwJG5YcF(cov07Iuzl(fNY7m7(dCFu4FgKrwV1rK49y2DeTNWCLkxkVFCjTfXeCwWW(ECSFZES1gEs8eFzgkDnxbS(Jf3ZtwfIBLfRvZ(xonZa)lbFo4Ojdv88LV39C2lnn0upcAzdog4QbLk0G6KhDkBxhAHvVlw2(476)cxhdcFTNlKIWSF9kwQeusB38fQ5dKldd6Um2Pd0lAlc5kb5ygUuv9hJ240igMO6I)IXE68tWZA6)sotGOK7N59ust(TmMsEa5TipK8Bi1)csdbImjoPjrbIevjOhCoslMfN0MOr(S39CedshcM4y)oJXbL0(Vr6soigXQpxDLkERTRrC3G4jqm1E6BtoeGF)(LVtEPnZwPsXs3A2YrlwIZrLbI1Z5SG34pLPUlXzttnKEBn3T)OToNamMGqoAQCTCgwYQ)b8Jbla0Y0RDMIHSJd4HuPDWOC5MoqMdd03k1XZWvF(VTbRzcAdhyTMqrBij18KVmUDc6YrXTHe77lt(9sBjTFXTiF58K)yJCuquuY5a(o6lNXYONJMtMwJHuWEoGyzvPhmqQivOmV2PWXlWfx2MMxYjoRkuvlpiPkeLo3WHP49EsEFSmBubN1LCODChpNo0o0LjxS7B(RmSuGIHY23I2EjSz1ChvL3W4GcSFA6BK6gxF1hclS(YIx5kIqKSRwaOvtV2kuqxNbI2rFlDmIdntQmRS61cdfOSmwSa8tWwU2QzUEA6wsNITNsOdfVTLLziWzOGZWbVTSrRrHZ4)QC4f8mm4WFs84CL7qQ4ouP4AtaW2ix)UYyneaWhmy1sH7pFFWIX2XIbT9VSyEODJKSfPNftURNPmM95s0o5dXbWN9dK6Wo)XWGTLDa)Vr5w10qfGc8oUtLvqUAOlRfxDzXBHCRXczlA2Yk3rf3AX0xFTRdMX0lnr6doT8kg62oOPs8SqRb77QBKSGUPk9JChrB6FXfcqCrgzwc0TaTCjXF3gIM6gH5iOai6WHUr4u4rOgo)Vc37XAiCeN5IJU3oYDd2xGlzS9ePh5uJams3Ttqggz7d6unvyTd(NINbSNTGb6oaPoQIkofnDwmx8xpr1iu7)TMm1U5yMOit0ob9zIK5kJsMjqHPOdd3oDm8xgNtnycAMffoX73qU4D0D8aRlpNh2jWAZhgoRDOb5MrevWSDtY)gYngq4Y(90pBcp(uL0vpK2mue2nIV7LsGmmJ34gZibdCRhx4PMS8Sl5yBO0KbZsn6L)mnvHR4t)uuuXnYeA0R4cwGEnhdNotrt2SnsDS1d8aJDEE4wcMrMjmdx223UuZFGqQWOReAGWH3tZKgXFO8sFI(fFCsOpz2zuG2qRmhkJvdr8GNaxAQXbzXAaBCypz5zWAiGyC1DaqwiWea(K4rF7jHV4PGKSd8XSaKjDhDJhqm(9loPEi(u)n9q)K8tlXsKKUVucnUJey9eRaU4sSBRZjOW)grBeGcB8kBasreXxIyoXL)aX0jxBa2bx0xe5DIP)JCmlY2hnnZy3K4KDAw2)VXNnHOMF6ICOlk9)gf8J)dZmhpSSgKei6zH4NOAvN3pDJMTGUMD0IZ)dCS5GjnXhWVRNFXOnQRBQ1dZUng2T5s13MYyjwYHT)i8x)V)iySdjzdBn5yNWUijhDvKeBAOtKn17WU2H1jNqN5V7ZrYoOQUy29u86Ng83sGDTVyheGUQZX09qzgPBUoY6Mmej)5xN0J8xi)v41))rxPptz2wxvfzkvkpmXL)kSHJ2a7FBksWzeD1MoOoGYkLoz6KP438unoCsnYvB0Z1sX3wCbH6F)c6Qpp8DTL7XwwDe63cdnec6bmkESxWhvTaDjQy3S6Ms5ZxsIrATmE6Q9xBZDUnER9(4D27AyDtk5rAnbLTLE749TW6GRNzh6TDzPIFs5s1YUBpQTGLB35VDzsQcKeXwWYoXnp(a)75T3WdSsSsnjsH5LICENltEwJJHQiGrfSSjC8AM86jdp1Eyks(QM8sRKNLif5Rzit(gow3ykyTWn3G8v9g2he5zlSIpU0BWjgG)Cu8xj5WleBeSjples07A6BSpFiu4xL1O48p4BLZiHydz0QtLrmeOCBugT6Sy06tsJwBgA06JRrRnDg9m6LCr524mA9zWOBmoJwpmJ6RBYVMcieAE7lO32ekO4pSnR1P80Rh0PrchPvsETKxlXbF39(pp

**Comment by InfusOnWoW** [comment]
You don't set changed to true in eihter onCastStart nor in onCastStop or onPowerUpdate.

You must set that to true, if there are changes in your state.

**Comment by justinryder** [comment]
Thank you very much for the quick reply! Adding that into my event handlers fixed it on the latest WeakAuras versions. :)

I'm still really confused why it would work without that in the version 3.7.6 or earlier of WeakAuras, which is why I submitted the bug in case other auras might've been affected by 3.7.7.

Looking at the diff again I see lots of changes related to `changed` checks, but they are kinda over my head as to how they actually affect aura triggers updating. Maybe the [GenericTrigger.lua changes](https://github.com/WeakAuras/WeakAuras2/compare/3.7.6...3.7.7#diff-205eaff132da5aff122af36254e3a519097ccf8a032bbbc9d084e5cb7d235b3c) could be related? Perhaps I was accidentally relying on a bug that was fixed by this version. Anyways, thanks again for helping me :D
